aten/src/ATen/native/native_functions.yaml

```diff
@@ -3634,10 +3634,16 @@
   dispatch:
     CPU: adaptive_avg_pool2d_out_cpu
     CUDA: adaptive_avg_pool2d_out_cuda
+    MkldnnCPU: mkldnn_adaptive_avg_pool2d_out
 
 - func: adaptive_avg_pool2d(Tensor self, int[2] output_size) -> Tensor
   python_module: nn
 
+- func: mkldnn_adaptive_avg_pool2d(Tensor self, int[2] output_size) -> Tensor
+  dispatch:
+    MkldnnCPU: mkldnn_adaptive_avg_pool2d
+  requires_tensor: True
```

Comment:
I might miss some context here. What's the meaning of this option?

Reply:
This makes the codegen script to insert code to unwrap a variable to a tensor before passing the input to the kernel:
https://github.com/pytorch/pytorch/blob/de19eee/tools/autograd/gen_variable_type.py#L869-L870
https://github.com/pytorch/pytorch/blob/de19eee/tools/autograd/gen_variable_type.py#L954